lib/nav_tab/link_collection.rb

```diff
@@ -0,0 +1,116 @@
+class NavTab::LinkCollection
+  include Rails.application.routes.url_helpers
+  include TranslationHelper
+
+  attr_reader :ability, :facility
+
+  def initialize(facility, ability)
+    @facility = facility
+    @ability = ability
+  end
+
+  def admin
+    default + [
+      admin_orders,
+      admin_reservations,
+      admin_billing,
+      admin_products,
+      admin_users,
+      admin_reports,
+      admin_facility,
+    ].select(&:present?)
+  end
+
+  def customer
+    default + [orders, reservations, accounts]
+  end
+
+  def default
+    [home]
+  end
+
+  private
+
+  def accounts
+    NavTab::Link.new(tab: :accounts, text: t_my(Account), url: accounts_path)
+  end
+
+  def admin_billing
+    ability.can?(:manage_billing, facility) &&
```

Comment:
Reformatted. Funny how the lines added up: RuboCop dinged this class for now being 101 lines long! (It prefers 100 or less)